outport/process/executionOrder/transactionsExecutionOrder.go

```diff
@@ -0,0 +1,258 @@
+	normalTxsHashes := make([][]byte, 0)
+	invalidTxsHashes := make([][]byte, 0)
+	for mbIndex, mb := range blockBody.MiniBlocks {
+		if isMBScheduled(header, mbIndex) == ignoreScheduled || shouldIgnoreProcessedMBScheduled(header, mbIndex) {
+			continue
+		}
+
+		isFromMe := mb.SenderShardID == header.GetShardID()
+		if !isFromMe {
+			continue
+		}
+
+		if mb.Type == block.TxBlock {
+			normalTxsHashes = append(normalTxsHashes, mb.TxHashes...)
+			continue
+		}
+		if mb.Type == block.InvalidBlock {
+			invalidTxsHashes = append(invalidTxsHashes, mb.TxHashes...)
+			continue
+		}
+	}
+
+	normalTxs := extractTxsFromMap(normalTxsHashes, pool.Txs)
+	invalidTxs := extractTxsFromMap(invalidTxsHashes, pool.Invalid)
+
+	return append(normalTxs, invalidTxs...)
+}
+
+func extractNormalTransactionAndScrsToMe(pool *outport.Pool, blockBody *block.Body, header data.HeaderHandler, ignoreScheduled bool) []data.TransactionHandlerWithGasUsedAndFee {
+	grouped := make([]data.TransactionHandlerWithGasUsedAndFee, 0)
+	for mbIndex, mb := range blockBody.MiniBlocks {
+		if isMBScheduled(header, mbIndex) == ignoreScheduled || shouldIgnoreProcessedMBScheduled(header, mbIndex) {
+			continue
+		}
+
+		isToMeCross := mb.ReceiverShardID == header.GetShardID() && mb.SenderShardID != mb.ReceiverShardID
+		if !isToMeCross {
+			continue
+		}
+
+		executedTxsHashes := extractExecutedTxHashes(mbIndex, mb.TxHashes, header)
+		if mb.Type == block.TxBlock {
+			grouped = append(grouped, extractTxsFromMap(executedTxsHashes, pool.Txs)...)
+			continue
+		}
+		if mb.Type == block.SmartContractResultBlock {
+			grouped = append(grouped, extractTxsFromMap(executedTxsHashes, pool.Scrs)...)
+			continue
+		}
+		if mb.Type == block.RewardsBlock {
+			grouped = append(grouped, extractTxsFromMap(executedTxsHashes, pool.Rewards)...)
+			continue
+		}
+	}
+
+	return grouped
+}
+
+func getRewardsTxsFromMe(pool *outport.Pool, blockBody *block.Body, header data.HeaderHandler) []data.TransactionHandlerWithGasUsedAndFee {
+	rewardsTxsHashes := make([][]byte, 0)
+	rewardsTxs := make([]data.TransactionHandlerWithGasUsedAndFee, 0)
+	if header.GetShardID() != core.MetachainShardId {
+		return rewardsTxs
+	}
+
+	for _, mb := range blockBody.MiniBlocks {
+		if mb.Type != block.RewardsBlock {
+			continue
+		}
+		isFromMe := mb.SenderShardID == header.GetShardID()
+		if !isFromMe {
+			continue
+		}
+		rewardsTxsHashes = append(rewardsTxsHashes, mb.TxHashes...)
+	}
+
+	return extractTxsFromMap(rewardsTxsHashes, pool.Rewards)
+}
+
+func extractTxsFromMap(txsHashes [][]byte, txs map[string]data.TransactionHandlerWithGasUsedAndFee) []data.TransactionHandlerWithGasUsedAndFee {
+	result := make([]data.TransactionHandlerWithGasUsedAndFee, 0, len(txsHashes))
+	for _, txHash := range txsHashes {
+		tx, found := txs[string(txHash)]
+		if !found {
+			log.Warn("extractTxsFromMap cannot find transaction in pool", "txHash", hex.EncodeToString(txHash))
```

Comment:
if this happens, somehow, it will break each and every index you compute afterward. Is this ok? I would return the error here and propagate it upwards.

Reply:
+1

Reply:
done